lib/spree_google_base/feed_builder.rb

```diff
@@ -83,7 +83,9 @@ def generate_xml(output)
           ar_scope.find_each() do |product|
             variants = Spree::Variant.find_each()
             variants.each do |variant|
-              if (variant.product_id == product.id && !variant.gtin.nil?) && (product.hide_for_customer == false && variant.hidden == false)
+              if (variant.product_id == product.id && !variant.gtin.nil?) &&
```

Comment:
i apologize for this shit storm of a method.... this was the first thing i worked on as an intern and it's utter garbage 😭 happy to help clean it up, it's so bad.

Reply:
Does it make any sense here to use some method extraction to name these conditionals? This might aid in speed of comprehension for the next person (maybe)?

Reply:
it could! and that's not a bad suggestion

for some additional context, in our next sprint i'm going to take a look at the level of effort involved in moving our product feed functionality into our glossier-v2 codebase rather than relying on this out-of-date forked gem

the rationale here is that we've already customized this gem quite a bit for our own (straightforward) use cases and we're not able to reliably test/spec this functionality as this codebase exists today (because the dependencies here i.e. solidus/spree, rspec, factorygirl/bot, etc. are pretty out-of-date)

very curious to hear your opinions as this work gets thought through!

Reply:
😍 this:

> look at the level of effort involved in moving our product feed functionality into our glossier-v2 codebase rather than relying on this out-of-date forked gem